concent_api/core/message_handlers.py

```diff
@@ -389,13 +389,14 @@ def handle_send_force_get_task_result(client_message: message.concents.ForceGetT
         task_to_compute,
     )
 
-    if Subtask.objects.filter(  # pylint: disable=no-member
-        subtask_id = task_to_compute.compute_task_def['subtask_id'],
-        state      = Subtask.SubtaskState.FORCING_RESULT_TRANSFER.name,  # pylint: disable=no-member
-    ).exists():
-        return message.concents.ServiceRefused(
-            reason = message.concents.ServiceRefused.REASON.DuplicateRequest,
-        )
+    try:
+        subtask = Subtask.objects.select_for_update().get(subtask_id=task_to_compute.compute_task_def['subtask_id'])
+        if subtask.state_enum == Subtask.SubtaskState.FORCING_RESULT_TRANSFER:
+            return message.concents.ServiceRefused(
+                reason=message.concents.ServiceRefused.REASON.DuplicateRequest,
+            )
```

Comment:
You should try to keep the `try ... except` part as small as possible to avoid catching unrelated exceptions.

This `if` should be in the `else:` clause:
``` python
try:
    # Get object
except ...:
    # Do something if it does not exist
else:
    # Do something with it if it exists
```

Reply:
ok, done. In this case I don't think it is necessary, because i catch only `Subtask.DoesNotExist`.  For me the old way was more readable... but changed :)

Reply:
Code changes. Now you have only this `if`. Later someone could add a function call. Then someone else comes and adds a database query in this function, not even realizing that it's used like this somewhere.

As a general rule it's safer to keep any extra code out of the `try ... except` block.